### ISSUE TITLE
Keras' PReLU is not channel-wise

### ISSUE THREAD

**Issue by crowsonkb** [fchollet/keras#3568]
According to the PReLU paper (https://arxiv.org/pdf/1502.01852v1.pdf section 2.1), "PReLU introduces a very small number of extra parameters. The number of extra parameters is equal to the total number of channels...". But in Keras PReLU (as well as the other parametric activations) is feature-wise, not channel-wise. This introduces a much larger number of additional parameters, when it should be adding half as many as an adjacent batch normalization layer.

Please make sure that the boxes below are checked before you submit your issue. Thank you!
- [x] Check that you are up-to-date with the master branch of Keras. You can update with:
  pip install git+git://github.com/fchollet/keras.git --upgrade --no-deps
- [x] If running on Theano, check that you are up-to-date with the master branch of Theano. You can update with:
  pip install git+git://github.com/Theano/Theano.git --upgrade --no-deps
- [ ] Provide a link to a GitHub Gist of a Python script that can reproduce your issue (or just copy the script here if it is short).

**Comment by the-moliver** [comment]
This is an excellent point. Their behavior is quite inconsistent, some like `LeakyReLU` use the same parameter across all features, some have a parameter per feature as you describe. They should all be modified for good defaults and maximum flexibility, probably something like how BatchNorm handles different normalization axes.

**Comment by ZiqianXie** [comment]
I am writing a similar channel-wise clip layer. I am using the theano backend, the shape of the weight is (1, channel, 1, 1) but the broadcast pattern of shared variable is default to false so I have to create auxiliary variable using K.T.addbroadcast in the function call to avoid the dimension mismatch problem.
